### PR TITLE
Added basic beatmap metadata impl

### PR DESCRIPTION
**Pull request by S1lentzer**
current code is quite meh, needs a bit of reworking

### CODE REVIEW

File: Rhythmic/Beatmap/BeatmapMetadata.cs [hunk 1]
```diff
@@ -11,5 +11,8 @@ public class BeatmapMetadata
         public string Description { get; set; }
         public int ID { get; set; }
         public TrackBass Track { get; set; }
+
+
```
Comment (AltenGD) [inline Rhythmic/Beatmap/BeatmapMetadata.cs]: remove useless blank spaces

File: Rhythmic/Beatmap/BeatmapsMetadata.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+
```
Comment (AltenGD) [inline Rhythmic/Beatmap/BeatmapsMetadata.cs]: Same applies to here

File: Rhythmic/Beatmap/BeatmapsMetadata.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace Rhythmic.Beatmap
+{
+    class BeatmapsMetadata
```
Comment (AltenGD) [inline Rhythmic/Beatmap/BeatmapsMetadata.cs]: Make this `public`

File: Rhythmic/Beatmap/BeatmapsMetadata.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            var lines = File.ReadAllLines("RhythmicDB.rh");
+            for (var i = 0; i < lines.Length; i += 1)
+            {
+                var line = lines[i];
+                ParseDatabase(line);
+            }
+        }
+    }
+
+
```
Comment (AltenGD) [inline Rhythmic/Beatmap/BeatmapsMetadata.cs]: Remove useless spacing

File: Rhythmic/Other/DrawableCarouselItem.cs [hunk 5]
```diff
@@ -9,11 +9,26 @@
 
 namespace Rhythmic.Other
 {
+
```
Comment (AltenGD) [inline Rhythmic/Other/DrawableCarouselItem.cs]: Remove useless spacing

File: Rhythmic/Other/DrawableCarouselItem.cs [hunk 6]
```diff
@@ -9,11 +9,26 @@
...
+        public string BeatmapName;
+        public string BeatmapAuthor;
+        public string SongName;
+        public string BGName;
+
+        public DrawableCarouselItem(string bname, string bauthor, string bgn)
+        {
+            BeatmapName = bname;
+            BeatmapAuthor = bauthor;
+            //SongName = songname;
```
Comment (AltenGD) [inline Rhythmic/Other/DrawableCarouselItem.cs]: If the code is commented, please remove them

File: Rhythmic/Screens/Menu/SongSelectCarousel.cs [hunk 7]
```diff
@@ -2,33 +2,42 @@
 using osuTK;
 using osu.Framework.Graphics;
 using Rhythmic.Other;
+using Rhythmic.Beatmap;
+using System.Collections.Generic;
+
```
Comment (AltenGD) [inline Rhythmic/Screens/Menu/SongSelectCarousel.cs]: Remove useless spacing

File: Rhythmic/Screens/Menu/SongSelectCarousel.cs [hunk 8]
```diff
@@ -2,33 +2,42 @@
...
                 }
             };
+            LoadBeatmaps();
+        }
+        public void LoadBeatmaps()
+        {
+            beatmaps.Load();
+            for (int i = 0; i < beatmaps.Entries; i++)
+            {
+                //Items.Add(new DrawableCarouselItem(beatmaps.Title[i], beatmaps.AuthorName[i], beatmaps.Backgrounds[i]) { Anchor = Anchor.Centre, Origin = Anchor.Centre });
```
Comment (AltenGD) [inline Rhythmic/Screens/Menu/SongSelectCarousel.cs]: If any code is commented, please remove them

**Comment by S1lentzer** [comment]
Closed until a moajor bug gets fixed.

**Pull request merged**